File: go/triangle/triangle.go
package triangle

import "math"

// Kind defines the kind of triangle
type Kind string

const (
	// NaT defines not a triangle
	NaT Kind = "Not a triangle"
	// Equ defines an equilateral triangle
	Equ Kind = "equilateral"
	// Iso defines an isosceles triangle
	Iso Kind = "isosceles"
	// Sca defines an scalene triangle
	Sca Kind = "scalene"
)

// KindFromSides should have a comment documenting it.
func KindFromSides(a, b, c float64) Kind {
	if math.IsNaN(a) ||
		math.IsNaN(b) ||
		math.IsNaN(c) ||
		math.IsInf(a, 1) ||
		math.IsInf(b, 1) ||
		math.IsInf(c, 1) {
		return NaT
	}

	if a*b*c <= 0 || c > a+b || a > b+c || b > a+c {
		return NaT
	}

	if a == b && b == c {
		return Equ
	}

	if a != b && b != c && c != a {
		return Sca
	}

	return Iso
}

File: python/triangle/triangle.py
def is_triangle(sides):
    return sides[0] + sides[1] + sides[2] > 0 and \
        sides[0] + sides[1] >= sides[2] and \
        sides[1] + sides[2] >= sides[0] and \
        sides[0] + sides[2] >= sides[1]

def equilateral(sides):
    return is_triangle(sides) and sides[0] == sides[1] == sides[2]

def isosceles(sides):
    return is_triangle(sides) and \
        (sides[0] == sides[1] or sides[0] == sides[2] or sides[1] == sides[2])

def scalene(sides):
    return is_triangle(sides) and \
        not equilateral(sides) and \
        not isosceles(sides)
